model-server/src/main/kotlin/org/modelix/model/server/store/ObjectInRepository.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2024.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.server.store
+
+import org.modelix.model.persistent.HashUtil
+
+data class ObjectInRepository(private val repositoryId: String, val key: String) {
+    fun isGlobal() = repositoryId == ""
+    fun isImmutable() = HashUtil.isSha256(key)
+    fun getRepositoryId() = repositoryId.takeIf { it != "" }
+
+    companion object {
+        /**
+         * Empty string is used for entries that don't belong to a repository, because PostgreSQL doesn't allow null
+         * values in primary keys.
+         */
+        fun global(key: String) = ObjectInRepository("", key)
```

Comment:
Can't we hide this implementation detail on some level below?

Reply:
Removed the comment. Now the reason is hidden in the implementation.

Reply:
Not really? The reason is hidden but the caller still needs to know the special meaning of an empty string in contrast to maybe `null`, which would be the more obvious choice without knowing the implications for the database.

Reply:
The empty string is not visible in the interface of this class. `getRepository()` returns `null` instead of an empty string and `ObjectInRepository.create` converts a null value to an empty string. I don't understand the problem.

Reply:
aaaah, the whole removed comment targeted the first argument of ObjectInRepository. I understood this to talk about `key`. Good that the confusing comment is gone then.